Classes/TextUtils.cs

```
@@ -24,19 +24,84 @@ namespace RadioDld
+            try
+            {
+                // Checks for match on ANY date in the string...
+                if (matchStripDate.IsMatch(name))
+                {
+                    // Date Found - now get the date value found
+                    string sDateFound = matchStripDate.Match(name).ToString();
+
+                    // Strip trouble characters eg ' and check for 'sept' and remove 't' also remove (st | nd | rd | th) and keep numeric
+                    sDateFound = sDateFound.Replace("'", string.Empty);
+                    sDateFound = sDateFound.Replace("Sept", "Sep").Replace("sept", "sep");
+                    sDateFound = removeDaySuffix.Replace(sDateFound, " ");
 
-            if (matchStripDate.IsMatch(name))
+                    // Convert to DateTime for comparison to similardate
+                    DateTime dtDateFound = Convert.ToDateTime(sDateFound, CultureInfo.CurrentCulture);
+
+                    // Check if date is within the +/- 'similardateday' range of 'date'
+                    if (dtDateFound >= stripDate.AddDays(-iSimilarDateDayRange) && dtDateFound <= stripDate.AddDays(iSimilarDateDayRange))
+                    {
+                       name = matchStripDate.Replace(name, string.Empty).ToString().Trim();
+                    }
+                }
+
+                return name;
+            }
+            catch
             {
-                name = matchStripDate.Match(name).Groups["name"].ToString();
+                return null;
             }
```

If you remove this try...catch which swallows the exception and returns null that should cause the exception occurring during two of the tests to be revealed.